### ISSUE TITLE
"Rational Numbers" exercise - divide_by function

### ISSUE THREAD

**Issue by kszambelanczyk** [exercism/elixir#891]
divide_by function according to the instructions:
`r₁ / r₂ = (a₁ * b₂) / (a₂ * b₁)`
implemented this way: 
```
def divide_by({n1, d1}, {n2, d2}) do
  {n1 * d2, d1 * n2}  |> reduce()
end

def reduce({n, d}) do
  {(n / Integer.gcd(n, d)), (d / Integer.gcd(n, d))}
end
```
for the test: 
```
test "Divide a positive rational number by a negative rational number" do
   r1 = {1, 2}
   r2 = {-2, 3}
   result = {-3, 4}
   assert RationalNumbers.divide_by(r1, r2) == result
end
```
returns:
{3, -4}
but test expects {-3, 4}
There is no information about switching the minus sign from denominator to numerator.

**Comment by jiegillet** [comment]
This is a standard thing when representing ratios. Your `reduce` function is also expected to put the minus sign in the denominator, even though it's not explicitly mentioned. 

In Haskell:
```haskell
Prelude> import Data.Ratio ((%))
Prelude Data.Ratio> 3 % (-4)
(-3) % 4
```

In Python
```python
>>> import fractions
>>> fractions.Fraction(3, -4)
Fraction(-3, 4)
```

In Julia
```Julia
julia> 3//-4
-3//4
```

This is a recurring theme in Exercism exercises, you are expected to figure out extra requirements from the tests. If you want to change the exercise description, you will have to open an issue [here](https://github.com/exercism/problem-specifications). But personally, I don't think this one is too bad, because again, it is very common for ratios.

**Comment by jiegillet** [comment]
We could add a specific test:

```elixir
test "Reduce shift the minus sign on the numerator" do
   r = {3, -4}
   result = {-3, 4}
   assert RationalNumbers.reduce(r) == result
end
```
Would you like to do a PR with that change?

**Comment by kszambelanczyk** [comment]
Yes. I would love to. Thank you.

**Comment by SleeplessByte** [comment]
That is actually a nice test to also add to problem-specifications.

**Comment by jiegillet** [comment]
Food idea, I proposed it [here](https://github.com/exercism/problem-specifications/issues/1833).
I've never contributed to `problem-specifications` before, the open PRs suggest that making a change takes 1000 years :p

**Comment by angelikatyborska** [comment]
@jiegillet not necessarily, it's just that the repository was "frozen" for a really long time because people tend to fight in it too much and we wanted some calm to build v3. Non-controversial changes get merged fast :D but you do need 3 approval.

**Comment by jiegillet** [comment]
Hahaha, let's see how this one goes :)

**Comment by jiegillet** [comment]
Closed via #892. Thank you for the new test!
